wikitexthtml/render/preprocess.py

```diff
@@ -43,6 +50,11 @@ def begin(instance: WikiTextHtml, body: str, is_transcluding: bool = False):
         start, end = match.span()
         body = body[:start] + "{{" + f"__snippet|post|{index}|nowiki" + "}}" + body[end:]
 
+    # If we don't want to color the syntax, remove all <syntaxhighlight> tags
+    # and replace them with <pre> tags.
+    if not should_color_syntax:
```

Comment:
I wonder if this is the most logical thing to do. Why not make another function, something like `syntax_highlight_as_pre`, which does this, and call that in an else in page.py `prepare`? Sending this to this function feels a bit like a hack?

Reply:
> Sending this to this function feels a bit like a hack?

We're parsing xml-like structures with regex -- I'd argue this has been in hack territory for a while. 

The issue is `<syntaxhighlight>` _looks_ like html, but it isn't -- so I believe passing it forward without styling would basically make it a `<div>` which would mean we wouldn't have preformatting at all.

We _could_ always force some sort of monospace font and CSS to put the section in a code block, but a nice thing about <pre> tags is that it leaves the specifics up to the browser. 

I am a _big_ fan of letting the user's browser make as many choices as possible. If browsers supported syntax highlighting natively, I'd hand it that over, as well. https://brutalist-web.design/

Reply:
On another read of your comment, I get the impression you're fine with `<pre>` tags but just want it handled outside of the `begin` function.

I'll push a change that does that and you can let me know if that looks better.